src/types/ethcall.d.ts

```diff
@@ -72,7 +75,12 @@ export class MulticallProvider extends Provider {
     D extends MulticallCall,
   >(
     calls: readonly [A, B, C, D],
+    canFail: boolean[],
     block?: number,
   ): Promise<[A["outputs"], B["outputs"], C["outputs"], D["outputs"]]>
-  tryEach<O>(calls: MulticallCall<I, O>[], block?: number): Promise<O[]> // fallthrough to array type
+  tryEach<O>(
```

Comment:
tbh I'm not sure how to type this off the top of my head, but now any call result may also be null if `canFail = true`.